cytominer_eval/utils/grit_utils.py

```diff
@@ -0,0 +1,167 @@
+    control_perts : list
+        The profile_ids that should be considered controls (the reference)
+    column_id_info: dict
+        A dictionary of column identifiers noting profile and replicate group ids. This
+        variable is autogenerated in
+        :py:func:`cytominer_eval.transform.util.set_grit_column_info`.
+    replicate_summary_method : {'mean', 'median'}, optional
+        how replicate z-scores to control perts are summarized. Defaults to "mean".
+
+    Returns
+    -------
+    dict
+        A return bundle of identifiers (perturbation, group) and results (grit score).
+        The dictionary has keys ("perturbation", "group", "grit_score"). "grit_score"
+        will be NaN if no other profiles exist in the defined group.
+    """
+    # Confirm that we support the provided summary method
+    check_grit_replicate_summary_method(replicate_summary_method)
+
+    group_entry = get_grit_entry(replicate_group_df, column_id_info["group"]["id"])
+    pert = get_grit_entry(replicate_group_df, column_id_info["profile"]["id"])
+
+    # Define distributions for control perturbations
+    control_distrib = replicate_group_df.loc[
+        replicate_group_df.loc[:, column_id_info["profile"]["comparison"]].isin(
+            control_perts
+        ),
+        "similarity_metric",
+    ].values.reshape(-1, 1)
+
+    assert len(control_distrib) > 1, "Error! No control perturbations found."
+
+    # Define distributions for same group (but not same perturbation)
+    same_group_distrib = replicate_group_df.loc[
+        (
+            replicate_group_df.loc[:, column_id_info["group"]["comparison"]]
+            == group_entry
+        )
+        & (replicate_group_df.loc[:, column_id_info["profile"]["comparison"]] != pert),
+        "similarity_metric",
+    ].values.reshape(-1, 1)
+
+    return_bundle = {"perturbation": pert, "group": group_entry}
+    if len(same_group_distrib) == 0:
+        return_bundle["grit"] = np.nan
+
+    else:
+        scaler = StandardScaler()
+        scaler.fit(control_distrib)
+        grit_z_scores = scaler.transform(same_group_distrib)
+
+        if replicate_summary_method == "mean":
+            grit_score = np.mean(grit_z_scores)
+        elif replicate_summary_method == "median":
+            grit_score = np.median(grit_z_scores)
+
+        return_bundle["grit"] = grit_score
+
+    return pd.Series(return_bundle)
+
+
+def get_grit_entry(df: pd.DataFrame, col: str) -> str:
+    """Helper function to define the perturbation identifier of interest
+
+    Grit must be calculated using unique perturbations. This may or may not mean unique
+    perturbations.
+    """
+    entries = df.loc[:, col]
+    assert (
+        len(entries.unique()) == 1
+    ), "grit is calculated for each perturbation independently"
+    return str(list(entries)[0])
+
+
+def set_grit_column_info(profile_col: str, replicate_group_col: str) -> dict:
+    r"""Transform column names to be used in calculating grit
```

Comment:
```suggestion
    """Transform column names to be used in calculating grit
```
don't really need that and it's inconsistent